tests/ops/qubit/test_observables.py

```diff
@@ -272,17 +272,31 @@ def test_hermitian_eigvals_eigvecs_same_observable_twice(
         assert len(qml.Hermitian._eigs) == 1
 
     @pytest.mark.parametrize("observable, eigvals, eigvecs", EIGVALS_TEST_DATA)
-    def test_hermitian_diagonalizing_gates(self, observable, eigvals, eigvecs, tol):
+    def test_hermitian_diagonalizing_gates(self, observable, eigvals, eigvecs, tol, mocker):
         """Tests that the diagonalizing_gates method of the Hermitian class returns the correct results."""
```

Comment:
The test failure I saw was in a different test. The CI logs are [here](https://github.com/PennyLaneAI/pennylane/actions/runs/5180782930/jobs/9335420469). I'm pretty sure there are a few others.

We can either introduce the same change to the rest of the failing tests, or create a new pytest fixture where we clean up `Hermitian._eigs` before/after every test.

Reply:
nice catch. if they're all scoped to this file, then it should be pretty straightforward. Just add something like this at the top of the file. don't even need to do stuff after! I'm doing it in another PR [here](https://github.com/PennyLaneAI/pennylane/pull/4257/files#diff-7388684292228c32fbc2732b7a087d9de8d7596d6c5b0836ad76c9fc5e04fc6aR28-R32). So:
```python
@pytest.fixture(autouse=True)
def run_before_tests():
    qml.Hermitian._eigs = {}
    yield
```

Reply:
Thanks! I'll add that :)